standards/SPIFFE_Workload_API.md

```diff
@@ -0,0 +1,170 @@
+
+When authenticating a client from a foreign trust domain, the authenticator chooses the bundle representing the client’s presented trust domain for validation. Similarly, when authenticating a server, the client uses the bundle representing the server’s trust domain. If no matching bundle is present for the SVID in use, then the peer is untrusted. This approach is required in order to account for the lack of widespread support for SAN URI Name Constraints in common X.509 libraries. Please see [Section 4.2](X509-SVID.md#42-name-constraints) of the X509-SVID specification for more information.
+
+### 5.3 Default Identity
+
+It is often the case that a workload doesn’t know what identity it should assume. Determining when to assume what identity is a site-specific concern, and as a result, the SPIFFE specifications don’t reason about how to do this.
+
+In order to support the widest variety of use cases, the X.509-SVID Profile supports the issuance of multiple identities, while also defining a default identity. It is expected that workloads which are aware of multiple identities can handle decision making on their own. Workloads which don’t understand how to leverage multiple identities may use the default identity. The default identity is the first in the list. Protocol buffers ensure that the order of the list is preserved.
+
+### 5.4 Profile Messages
+
+The X.509-SVID Profile messages are expressed as a Protocol Buffer version 3 (proto3). They are defined below. For the specific service definition for this profile, please see the SPIFFE Workload API [Service Definition](#3-service-definition).
+
+```protobuf
+// The X509SVIDResponse message carries a set of X.509 SVIDs and their
+// associated information. It also carries a set of global CRLs, and a
+// TTL to inform the workload when it should check back next.
+message X509SVIDResponse {
+    // A list of X509SVID messages, each of which includes a single
+    // SPIFFE Verifiable Identity Document, along with its private key
+    // and bundle.
+    repeated X509SVID svids = 1;
+
+    // ASN.1 DER encoded
+    repeated bytes crl = 2;
+
+    // CA certificate bundles belonging to foreign Trust Domains that the
+    // workload should trust, keyed by the SPIFFE ID of the foreign
+    // domain. Bundles are ASN.1 DER encoded.
+    map<string, bytes> federated_bundles = 3;
+}
+
+// The X509SVID message carries a single SVID and all associated
+// information, including CA bundles.
+message X509SVID {
+    // The SPIFFE ID of the SVID in this entry
+    string spiffe_id = 1;
+
+    // ASN.1 DER encoded certificate chain. MAY include intermediates,
+    // the leaf certificate (or SVID itself) MUST come first.
+    bytes x509_svid = 2;
+
+    // ASN.1 DER encoded PKCS#8 private key. MUST be unencrypted.
+    bytes x509_svid_key = 3;
+
+    // CA certificates belonging to the Trust Domain
+    // ASN.1 DER encoded
+    bytes bundle = 4;
```

Comment:
Placing the "native" bundle in the X509SVID message complicates validation, since the validator must understand when to consult this bundle or the federated bundles map. Feels like it makes more sense to put _all_ bundles in federated_bundles, including those belonging to issued SVIDs... thoughts? Maybe that means federated_bundles -> bundles?

Reply:
I had been assuming that this is basically the certificate chain that would be presented during a TLS handshake. Whereas the certificates in `federated_bundles` would be the self-signed roots of the foreign trust domain; at least for the purposes of TLS, including subordinate CA certs from a foreign trust domain isn't necessary since a foreign server will send those intermediate certs during the handshake.

If we're on the same page here about the intended usage of `bundle`, then perhaps we should add a comment to clarify (something like: `bundle` should include all intermediate certificates needed during a TLS handshake).

Reply:
To address your concern about the validator, we could add another field like `string trust_domain`. Not sure it's really justified though, a client should be able to pull the trust domain out of either `spiffe_id` or `x509_svid`.

Reply:
>I had been assuming that this is basically the certificate chain that would be presented during a TLS handshake.

The chain (if any) is currently included in the `x509_svid` field (there is a comment there to that effect). This allows the service provider to control the presented chain. The `bundle` field must include the trust domain roots so that the service can authenticate presented certificates in its own trust domain, and if it additionally provided the intermediates, then it would be up to the workload to decide how many of the intermediates to present with the leaf... that feels like a slippery slope IMO.

>To address your concern about the validator, we could add another field like string trust_domain. 

My comment here on possibly moving the trust domain roots to `federated_bundles` is geared towards avoiding an extra if/else in client code. The way it is currently, client validation has to have something like:
```go
if spiffe.TrustDomain(clientSVID) == myTrustDomain {
     // Validate using certs from `bundle`
} else {
     // Validate using certs from `federated_bundle`
}
```
By having `federated_bundle` be the default (only) case, I think we can 1) slightly simplify the validation logic, and 2) lean towards federation being the norm rather than the exception.